webgpu.c

```diff
@@ -180,25 +216,43 @@ void wgpuAdapterRelease(WGPUAdapter adapter)
 // {
 // }
 
-// void wgpuBindGroupAddRef(WGPUBindGroup bindGroup)
-// {
-// }
+void wgpuBindGroupAddRef(WGPUBindGroup bindGroup)
+{
+    if(!bindGroup) unreachable();
+    bindGroup->refCount ++;
+}
 
-// void wgpuBindGroupRelease(WGPUBindGroup bindGroup)
-// {
-// }
+void wgpuBindGroupRelease(WGPUBindGroup bindGroup)
+{
+    if(!bindGroup) unreachable();
+    bindGroup->refCount --;
+    if(bindGroup->refCount < 1)
+    {
+        wasi_webgpu_webgpu_gpu_bind_group_drop_own(bindGroup->bind_group);
+        free(bindGroup);
+    }
```

Comment:
We should trap to catch a real end-user bug where Release was called more than AddRef. 
```suggestion
    if(bindGroup->refCount == 0) {
        unreachable();
    }
    bindGroup->refCount --;
    
    wasi_webgpu_webgpu_gpu_bind_group_drop_own(bindGroup->bind_group);
    free(bindGroup);
```

Reply:
Added check to all release functions